### DISCUSSION TITLE
make the dataset featureless

### DISCUSSION THREAD

**Discussion by cxw-droid** [pyg-team/pytorch_geometric#7616] [Q&A] [answered]
Hi,

How can I make a dataset (e.g,. MUTAG) featureless (e.g. make all features be `1`) after I get the dataset from TUDataset(). When I tried to set each graph Data by `data.x = torch.Tensor([1]*len(data.x))`, the original graph `Data.x` is not changed, and when I tried to create a new list of graphs Data, the new list has no properties such as `num_features`.

So my questions is: Is there a convenient way to transform the default MUTAG dataset to a featureless one, at the same time keeping the original properties like `num_features` etc? Thanks.

**Comment by rusty1s** [comment, marked as answer]
```python
transform = T.Constant(cat=False)
dataset = TUDataset(path, name='MUTAG', transform=transform)
